### DISCUSSION TITLE
Refill error on resume after refilled

### DISCUSSION THREAD

**Discussion by Fellstone7** [Guilouz/Creality-Helper-Script-Wiki#474] [Issues]
Hi I have the latest Script and firmware install and ran into this issue today. Any recommendations..

14:44
{"code":"key111", "msg": "Extrude below minimum temp
14:44
{"code":"key111", "msg": "Extrude below minimum temp
See the 'min_extrude_temp' config option for details", "values": []}
14:44
Restarting print...

**Comment by TheWebMachine** [comment]
I just had this happen last night and ran into the same exact issue. Ruined my print. I was going to do a second test to confirm before posting, but you beat me to it. 🫡

Are you using Creality or Guppy screen software?

**Reply by Fellstone7** [reply]
No I set the print off on the K1 Max screen. I am just using the standard Creality UI. Normally on the previous 1.3.3.5 and the pre 5 script it just worked if I ran out. Same as you I lost the print. I have just done another this time setting it off through the Mainsail interface. The print is two colour so I have a pause 70% of the way through. This worked, no indication or prompt however. I just change filiment and resumed via the Mailsail interface and it did just work. So not sure if setting the print on on the Standard Creality machine screen was the issue or restarting it on the main screen. Weird...

**Reply by TheWebMachine** [reply]
I had the issue while using Guppy and I resumed from the screen after a filament out condition, not a M600 color change. Same thing, though, the extruder failed to heat back up after loading in the filament. It heated up for that but then cooled down between filament load and resume. Something isn't retaining/restoring the print job hotend temp, I suppose.

**Reply by Fellstone7** [reply]
This was a filament out condition just not on the Guppy screen. I suspect this could be a Creality 1.3.3.8 issue....

**Comment by Fellstone7** [comment]
The M600 worked ok.

**Comment by TheWebMachine** [comment]
Add this to the Creality screen freeze issue, continued wifi instability, and a couple other random issues, and it looks like Creality has, yet again, reached their goal of fixing nothing noteworthy while breaking 3 other things. 🤦🏻‍♂️ I really should stop being surprised by this...

I'm rolling back to 1.3.3.5 tonight. .8 is a total mess.

**Comment by halTWO3** [comment]
My experience is the complete opposite. Aside from screen freezing (that's a prusaslicer issue, at least for me), I'm about to finish a 14hr print and prior to that a 16hr print using the latest firmware without issues. No color changes though.

**Reply by TheWebMachine** [reply]
Is the screen freezing due to Prusaslicer? Any idea why? I didn't starting having the screen freeze during prints until 1.3.3.8 and confirmed via rollback that I do *not* have the issue in 1.3.3.5. 🧐

**Reply by TheWebMachine** [reply]
Oh, and so long as there are no filament changes or runouts, all my prints finish, too. My major issues are:

- Wifi has been funky since day one, both FW versions. I live in a dense suburban area and 2.4ghz spectrum is filled to the brim. It seems if you use a channel other than 1, 6, or 11, the wifi becomes unstable at *any* distance from the AP; a mere 3 meters/10ft in my case! Using 1, 6 or 11 is difficult for me here because those are defaults for most routers and most folks never change it. I had been using channels 3 and 9 for a long time in my house and only this new K1C had a problem with it. Upon switching the closer AP to Ch1, despite being far more crowded by my neighbors, the wifi has been stable, albeit slower due to the congestion. I'm so sick of new devices only being 2.4G. 🤬

- Screen freeze during print only since 1.3.3.8

- M600 or pause/resume since 1.3.3.8 doesn't keep hotend at temp when resuming print. It will heat for the filament swap, then cool down before attempting final purge and resume of print.

It would be a huge help if Creality's changelog was a little more verbose than "some bug fixes." I hate that lazy BS. How the hell is anyone supposed to troubleshoot for breaking changes when they won't tell us what they broke. 🤦🏻‍♂️

**Reply by halTWO3** [reply]
When using default ui, my screen freezes instantly when a print is started. I think it's coz of the way prusaslicer handles thumbnail generation. Issue started after the last fw update, which started to support  object exclusion by physical display screen (installing guppyscreen removes this feature). No issues when using orcaslicer.

In regards to wifi, we are in the same situation. Had to set ap to cycle channels every 3 days. However, I only started getting timeouts since fw v1.3.3.5.  I was rolling fw v1.3.2.0 without any of these issues. Printer is only 3-4 months old,  over 1400+ hrs print time already.

I use a different printer for color swaps.  Didn't wanna bother doing it with the K1. I don't like how the filament is inserted, keeps getting stuck by the extruder. Plus, I do long prints, at least 14-20hrs+. Don't wanna swap at 80% and fail.

**Comment by SnifMyBack** [comment]
I've just ran into the same problem today in a middle of a print. The M600 command went fine but when the macro was triggered by the filament run out sensor, this is the error it throwed:  
{"code":"key111", "msg": "Extrude below minimum temp See the 'min_extrude_temp' config option for details", "values": []}

**Reply by lee-kev** [reply]
I had this same exact issue today.  I could re-fill the filament and pressed resume which then errored out.

```
11:43:23 // z_safe = 44.52594296354415
11:43:23 // fan2_value = 0
11:43:31 // Run Current: 0.28A Hold Current: 0.28A
11:43:31 // z_safe = 2.0
11:43:31 // Print paused for filament change!
11:43:31 // {"code":"key211", "msg": "Print already paused", "values": []}
11:43:34 // Extracting filament...
11:43:39 // idle_timeout: Timeout set to 900.00 s
11:43:53 // Run Current: 0.28A Hold Current: 0.28A
11:43:53 // Replace filament at the extruder inlet and click on Resume button!
11:57:54 // Run Current: 0.56A Hold Current: 0.56A
11:58:45 // Run Current: 0.28A Hold Current: 0.28A
11:59:02 // Run Current: 0.56A Hold Current: 0.56A
11:59:03 // Starting hotend heating...
11:59:09 // Loading and purging filament...
12:00:09 // idle_timeout: Timeout set to 99999999.00 s
12:00:09 // Restarting print...
12:00:10 // {"code":"key111", "msg": "Extrude below minimum temp
// See the 'min_extrude_temp' config option for details", "values": []}
12:00:10 !! {"code":"key111", "msg": "Extrude below minimum temp
```
The K1Max touch screen then went back to the main menu with the popup error about something that I forgot to capture.

Running v1.3.3.5 helper-script v5.0.1

**Reply by TheWebMachine** [reply]
It's the same on the K1C. The popup you missed is basically just the same error about the extruder temp being too low.

The macros are not restoring the hotend temp of the print job after the load routine. As best I can tell, it starts dropping temp after the initial load purge, before the resume print purge. My extruder started to skip towards the end of the second purge routine; the one that happens once you tell it to resume the print. So, probably the resume macro is missing the mark. I haven't dug into those macros yet, myself.

**Reply by Fellstone7** [reply]
Thats exactly the problem I encountered.

**Reply by TheWebMachine** [reply]
Then I think we know where to start looking! Maybe I can find some time to look into the M600 and related macros tonight. Thanks for the details and confirmation!

**Reply by Guilouz** [reply]
Can you try latest script update. I have added a safety check of extrudeur temp.

**Comment by Danmor74** [comment]
What do we do to apply the fix for this? I see there is a script update but can we just apply a fix to the M600 resume issue?

**Reply by halTWO3** [reply]
I think the best way is update script then remove/reinstall m600 option using script menus.

Edit: As the script author said, just reboot!

**Reply by Guilouz** [reply]
You just need to update script and reboot to take effect.

**Comment by lee-kev** [comment]
Updated script, reloaded printer and ran into the same issue.

**Reply by lee-kev** [reply]
![image](https://github.com/Guilouz/Creality-K1-Series/assets/104095237/13cd8bec-b9f7-4228-a889-70e84526bcaf)

`
19:24:02 // z_safe = 37.72509692305335
19:24:02 // fan2_value = 0
19:24:08 // Run Current: 0.28A Hold Current: 0.28A
19:24:09 // z_safe = 2.0
19:24:09 // Print paused for filament change!
19:24:09 // {"code":"key211", "msg": "Print already paused", "values": []}
19:24:10 // Extracting filament...
19:24:16 // idle_timeout: Timeout set to 900.00 s
19:24:30 // Run Current: 0.28A Hold Current: 0.28A
19:24:30 // Replace filament at the extruder inlet and click on Resume button!
19:39:33 // Stopping hotend heating...
20:54:29 // Run Current: 0.56A Hold Current: 0.56A
20:55:45 // Run Current: 0.28A Hold Current: 0.28A
20:56:41 // Run Current: 0.56A Hold Current: 0.56A
20:56:41 // Starting hotend heating...
20:56:42 // Starting hotend heating...
20:57:25 // Extruder not hot enough!
20:57:25 // idle_timeout: Timeout set to 99999999.00 s
20:57:25 // Restarting print...
20:57:25 // {"code":"key111", "msg": "Extrude below minimum temp
// See the 'min_extrude_temp' config option for details", "values": []}
20:57:25 !! {"code":"key111", "msg": "Extrude below minimum temp
`

**Reply by Guilouz** [reply]
Send your klipper.log

**Reply by Danmor74** [reply]
Yes I did the same thing.
Ran the update via SSH after logging in to the printer. The script asked me to run the update.
I then rebooted the Printer.

I started a print and let it go for the first layer, then removed the filament from the runout sensor and it paused asking me to refill as normal.

I did my usual retract and then put the filament back in the runout sensor and ran extrude as normal.

On the K1 pad I then go to home and pressed start and resume print.

the nozzle temperature set point goes to 140 so I then had to wait for it to hit that temp. When it gets to that temperature it then starts to purge which obviously just grinds the filament, it then tries to start printing immediately after while it’s trying to heat up and then faults on under temp.

I replicated the same behaviour resuming from mainsail.

I then just tried pausing a print and then resuming and that worked as it should.

it seems as though it is setting the temperature to the pause temperature on resume when it should be setting it back to printing temperature.

**Reply by Danmor74** [reply]
I just removed M600 support using the script and now it works as it should again. It doesn't seem to be a 1.3.3.8 issue unless they changed the way it works in the firmware.
I don't need to set filament changes anyway and if I do need to I can just set a pause and go through the process myself.
I can't afford to not have the runout sensor as my printers run 24/7 and I have lost a few prints over the past couple of days that were 60-70% through.

**Comment by Schulli73** [comment]
I have the same problem 

and have another problem, I see it here in the console 12X 

14:44
Command {v4l2-ctl} finished
14:44
Running Command {v4l2-ctl}...:
Can someone tell me what this is and how to fix it

**Comment by lee-kev** [comment]
Just wanted to note that the M600 script is working as intended now.  I was able to refill my filament.

**Comment by TheWebMachine** [comment]
I'll give it another try this evening with both an "oops" run-out and a gcode M600 filament change.
